src/Kunstmaan/MenuBundle/Twig/MenuTwigExtension.php

```diff
@@ -50,44 +51,30 @@ public function getFunctions()
      */
     public function getMenu($name, $lang, $options = array())
     {
-        $repo = $this->em->getRepository('KunstmaanMenuBundle:MenuItem');
-        $query = $this->em
-            ->createQueryBuilder()
-            ->select('mi, nt, p')
-            ->from('KunstmaanMenuBundle:MenuItem', 'mi')
-            ->innerJoin('mi.menu', 'm')
-            ->leftJoin('mi.parent', 'p')
-            ->leftJoin('mi.nodeTranslation', 'nt')
-            ->leftJoin('nt.node', 'n')
-            ->orderBy('mi.lft', 'ASC')
-            ->where('m.locale = :locale')
-            ->setParameter('locale', $lang)
-            ->andWhere('m.name = :name')
-            ->setParameter('name', $name)
-            ->andWhere('(nt.online = 1 OR nt.online IS NULL)')
-            ->andWhere('(n.deleted = 0 OR n.deleted IS NULL)')
-            ->getQuery();
-        $arrayResult = $query->getArrayResult();
-
-        // Make sure the parent item is not offline
-        $foundIds = array();
+	/** @var MenuItem $menuRepo */
+	$arrayResult = $this->repository->getMenuItemsForLanguage($name, $lang);
+
+	// Make sure the parent item is not offline
+	$foundIds = array();
         foreach ($arrayResult as $array) {
             $foundIds[] = $array['id'];
         }
         foreach ($arrayResult as $key => $array) {
-            if (!is_null($array['parent']) && !in_array($array['parent']['id'], $foundIds))  {
+	    if (!is_null($array['parent']) && !in_array($array['parent']['id'], $foundIds)) {
                 unset($arrayResult[$key]);
             }
         }
 
+	// SET OPTIONS
+	// Class for active item
         $activeClass = false;
-
         if (isset($options['activeClass'])) {
             $activeClass = $options['activeClass'];
         }
 
         $options = array_merge($this->getDefaultOptions($activeClass), $options);
-        $html = $repo->buildTree($arrayResult, $options);
+
+	$html = $menuRepo->buildTree($arrayResult, $options);
 
```

Comment:
I think $menuRepo either needs to be defined before this point, or you need to use $this->repository instead